master/buildbot/statistics/capture.py

```diff
@@ -0,0 +1,200 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+
+from twisted.internet import defer
+from twisted.internet import threads
+
+
+class Capture(object):
+    """
+    Base class for all Capture* classes.
+    """
+    def __init__(self, routingKey, callback):
+        self.routingKey = routingKey
+        self.callback = callback
+        # parent service and buildmaster to be set when StatsService initialized
+        self.parent_svcs = []
+        self.master = None
+
+    def defaultContext(self, msg):
+        return {
+            "builder_name": self.builder_name,
+            "build_number": str(msg['number'])
+        }
+
+    def consumer(self, routingKey, msg):
```

Comment:
I personally think that we should just use `NotImplementedError`. It would be more idiomatic Python since Python follows more of a 'easier to ask for forgiveness than permission' principle.